lib/galaxy/managers/histories.py

```diff
@@ -955,28 +1028,33 @@ def get_custom_builds_metadata(self, trans, id: EncodedDatabaseIdField) -> Custo
             fasta_hdas=[LabelValuePair(label=f'{hda.hid}: {hda.name}', value=trans.security.encode_id(hda.id)) for hda in fasta_hdas],
         )
 
-    def sharing(self, trans, id: EncodedDatabaseIdField, payload: Optional[sharable.SharingPayload] = None) -> sharable.SharingStatus:
+    def sharing(
+        self,
+        trans: ProvidesUserContext,
+        id: EncodedDatabaseIdField,
+        payload: Optional[sharable.SharingPayload] = None,
+    ) -> sharable.SharingStatus:
         """Allows to publish or share with other users the given resource (by id) and returns the current sharing
         status of the resource.
         """
         return self.shareable_service.sharing(trans, id, payload)
 
     def _serialize_history(
             self,
-            trans,
+            trans: ProvidesHistoryContext,
             history: model.History,
             serialization_params: SerializationParams,
             default_view: str = "detailed",
-    ) -> Union[HistoryBeta, HistoryDetailed, HistorySummary]:
+    ) -> Union[AnyHistoryView]:
```

Comment:
This is already a union right? Should this just be ``AnyHistoryView``?

Reply:
Good catch! that's just a bad replace operation :sweat_smile: